dialogue-core/src/main/java/com/palantir/dialogue/core/LimitedResponse.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.dialogue.Response;
+import org.derive4j.Data;
+
+@Data
+public interface LimitedResponse {
```

Comment:
This appears to only be used by RoundRobinChannel, is that correct?

Reply:
No, all selection strategies convert from LimitedReponses to Optional<Response>. The idea is that selection strategies need more information to properly select which channel to make a request to